src/main/java/com/atlassian/db/replica/internal/ReplicaConnectionProvider.java

```diff
@@ -189,9 +190,16 @@ public void commit() throws SQLException {
     private void recordCommit(boolean autoCommit) throws SQLException {
         if (state.getState().equals(MAIN) && !autoCommit) {
             consistency.write(state.getWriteConnection(new RouteDecisionBuilder(Reason.RW_API_CALL)));
+            state.clearDirty();
```

Comment:
it's cleared only when ` consistency.write` happen, not always-> outside of check? 
I guess then we would do it more excessively.

Reply:
Yes `consistency.write` means we committed the changes and we track it with ReplicaConsistency.